cdap-examples/Purchase/src/test/java/co/cask/cdap/examples/purchase/PurchaseAppTest.java

```diff
@@ -142,6 +145,18 @@ private ServiceManager getServiceManager(ApplicationManager appManager) throws I
     return userProfileServiceManager;
   }
 
+  private String getServiceUrlWithRetry(ServiceManager serviceManager) {
+    int trial = 0;
+    while (trial++ < 5) {
+      URL url = serviceManager.getServiceURL(15, TimeUnit.SECONDS);
+      if (url != null) {
```

Comment:
There is no "empty" for URL. It shows as null because `url == null` and when doing string concat, it turns in to `"null"`.